File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mapa = {}
        for i, v in enumerate(nums):
            complemento = target - v
            if complemento in mapa:
                return [mapa[complemento], i]
            mapa[v] = i
        return []

File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int izq=0, der=numbers.size()-1;
        vector<int> sol;
        while(izq<der){
            if(numbers[izq]+numbers[der] == target)
                return {izq+1, der+1};
            else if(numbers[izq]+numbers[der] > target)
                der--;
            else 
                izq++;
        }
        return {};
    }
};
